tests/vegalite/v5/test_theme.py

```diff
@@ -23,10 +36,952 @@ def test_vega_themes(chart) -> None:
 
 def test_register_theme_decorator() -> None:
     @register_theme("unique name", enable=True)
-    def custom_theme() -> dict[str, int]:
+    def custom_theme() -> ThemeConfig:
         return {"height": 400, "width": 700}
 
     assert themes.active == "unique name"
     registered = themes.get()
     assert registered is not None
     assert registered() == {"height": 400, "width": 700} == custom_theme()
+
+
+@pytest.mark.parametrize(
+    ("color_code", "valid"),
+    [
+        ("#FFFFFF", True),
+        ("##ff6347", False),
+        ("#EE82EE", True),
+        ("#1ec9a0", True),
+        ("#19B_71", False),
+        ("#00#761", False),
+        ("123455", False),
+        ("#6a5acd ", False),
+        ("#f8f8f899", True),
+        ("#6a5acd6E", True),
+    ],
+)
+def test_is_color_hex(color_code: Any, *, valid: bool) -> None:
+    assert is_color_hex(color_code) == valid
+
+
+def carbonwhite_theme() -> ThemeConfig:
+    """
+    Only including **1/4** of `carbon`_ , which gives sufficient structural coverage.
+
+    .. _carbon:
+        https://github.com/vega/vega-themes/blob/5f1a5c5b22cc462cf3d46894212152b71cfe964f/src/carbongen.ts
+    """
+    return ThemeConfig(
+        config={
+            "arc": {"fill": "#6929c4"},
+            "area": {"fill": "#6929c4"},
```

Comment:
Noting here that `mypy` runs are now **significantly** longer, due to all the inference required for `test_theme.py`.

I think all of this is worth it right now for completeness, but if this is a concern I can trim these tests down

Reply:
Really looking forward to when [`ruff` adds `mypy` compatible linting](https://github.com/astral-sh/ruff/issues/3893#issuecomment-2119208889) :)